From 9d912ae4c1e5baaccd3d8e25b5e4bd50b01661c4 Mon Sep 17 00:00:00 2001
From: me-cafebabe <me.cafebabe@gmail.com>
Date: Wed, 13 Sep 2023 16:29:10 +0800
Subject: [PATCH] partition: Wipe_F2FS: Only enable casefolding for /data

* Reference: https://github.com/LineageOS/android_bootable_recovery/blob/e5c3f128ca728ae53c1f9606e79f85df817bdeee/recovery_utils/roots.cpp#L234

Change-Id: Ia1f450cbd3b263b57f4a6a77e54835a5e09ca07f
(cherry picked from commit d2c43aeae5274be89998c179a248f5ba6a959f91)
(cherry picked from commit d90a63b81ac6441cc892ca6f23e8c6a031a8062b)
---
 partition.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/partition.cpp b/partition.cpp
index d7ab3b58..2ee61e27 100755
--- a/partition.cpp
+++ b/partition.cpp
@@ -2500,7 +2500,10 @@ bool TWPartition::Wipe_F2FS() {
 		return false;
 	}
 
-	needs_casefold = android::base::GetBoolProperty("external_storage.casefold.enabled", false);
+	if (Mount_Point == "/data") {
+		needs_casefold = android::base::GetBoolProperty("external_storage.casefold.enabled", false);
+	}
+
 	unsigned long long dev_sz = TWFunc::IOCTL_Get_Block_Size(Actual_Block_Device.c_str());
 	if (!dev_sz)
 		return false;
-- 
2.53.0

